packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -284,10 +287,128 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
             start: xPos,
             end: xPos
         };
+        this.xStartPos = xPos;
         document.addEventListener('mousemove', this.updateSelection);
         document.addEventListener('mouseup', this.endSelection);
     }
 
+    private zoomIn(diffStartMouse: number, diffEndMouse: number, percentRange: number) {
+        if (this.isRangeMove) {
+            this.props.unitController.viewRange = {
+                start: this.props.unitController.viewRange.start + percentRange,
+                end: this.props.unitController.viewRange.end - percentRange
+            };
+        } else {
+            const newStartRange = this.xStartPos - (diffStartMouse - (diffStartMouse * 0.1));
+            const newEndRange = this.xStartPos + (diffEndMouse - (diffEndMouse * 0.1));
+            this.props.unitController.viewRange = {
+                start: newStartRange,
+                end: newEndRange
+            };
+        }
+    }
+
+    private zoomOut(diffStartMouse: number, diffEndMouse: number, percentRange: number) {
+        if (this.isRangeMove) {
+            this.props.unitController.viewRange = {
+                start: this.props.unitController.viewRange.start - percentRange,
+                end: this.props.unitController.viewRange.end + percentRange
+            };
+        }
+        else {
+            const newStartRange = this.xStartPos - (diffStartMouse + (diffStartMouse * 0.1));
+            const newEndRange = this.xStartPos + (diffEndMouse + (diffEndMouse * 0.1));
```

Comment:
Does xy chart zooming and timegraph chart zooming behave the same way? I think the zoom factor used is different. 
Refer to this: https://github.com/theia-ide/timeline-chart/blob/2d3f4d88d8f5f214b5b76136ccdfa25d2d645297/timeline-chart/src/layer/time-graph-chart.ts#L81

Reply:
Should I use the same zoom factor? When I tried to zoom in and out in the timeline view, the zoom in and out is sometimes larger than I want.

Reply:
Okay. I am not sure about that. It would be nice to keep the zooming consistent across the charts. I think we can create a new issue for that.